### DISCUSSION TITLE
Workflow on how to add external self written adapter into streampipes

### DISCUSSION THREAD

**Discussion by flomickl** [apache/streampipes#1381] [Dev] [answered]
If I write my own adapter that cannot be included into StreamPipes due to license issues;
What would be the workflow to add these external adapters into StreamPipes so they are available in the pipeline editor?
I mean in CLI start or even the usual installer?

This solution was mentioned in another discussion here https://github.com/apache/streampipes/discussions/1034#discussioncomment-4595681

> Hi Florian, while this would certainly be a great adapter, unfortunately we can't directly add the adapter to StreamPipes due to license issues (API is free for research and non-commercial usage) [1].
> 
> We could however work on supporting the key value format and you could, for instance, create a specific adapter in your own repository and maybe we could also add references to "unsupported community-created extensions" to the website.
> 
> [1] https://openskynetwork.github.io/opensky-api/index.html

**Comment by dominikriemer** [comment, marked as answer]
Hi @flomickl it should be very similar to registering pipeline elements - just that in contrast to pipeline elements, adapters don't need to be installed in the UI and are automatically available in the adapter overview.
See, e.g., `streampipes-connect-adapters-iiot` for some examples.
